src/highdicom/sr/content.py

```diff
@@ -599,3 +838,163 @@ def __init__(
+        # referenced frames
+        refd_insuids = set()
+        for frame_info in referenced_frame_info:
+            if hasattr(frame_info, 'DerivationImageSequence'):
+                for drv_image in frame_info.DerivationImageSequence:
+                    if hasattr(drv_image, 'SourceImageSequence'):
+                        for src_image in drv_image.SourceImageSequence:
+                            # Check to avoid duplication of instances
+                            ins_uid = src_image.ReferencedSOPInstanceUID
+                            cls_uid = src_image.ReferencedSOPClassUID
+                            if ins_uid not in refd_insuids:
+                                if hasattr(src_image, 'ReferencedFrameNumber'):
+                                    ref_frames = src_image.ReferencedFrameNumber
+                                else:
+                                    ref_frames = None
+
+                                source_images.append(
+                                    SourceImageForSegmentation(
+                                        referenced_sop_class_uid=cls_uid,
+                                        referenced_sop_instance_uid=ins_uid,
+                                        referenced_frame_numbers=ref_frames
+                                    )
+                                )
+                                refd_insuids |= {ins_uid}
+
+        if len(source_images) == 0:
+            # Attempt to retrieve the source series instance uid from the
+            # dataset
+            # Note since only a single series may be used, we take the
+            # first
+            # ReferencedSeriesSequence is a type 1C that should be
+            # present in all segmentations that reference other instances
+            # within the study
+            if hasattr(dataset, 'ReferencedSeriesSequence'):
+                ref_series = dataset.ReferencedSeriesSequence[0]
+                if hasattr(ref_series, 'ReferencedInstanceSequence'):
+                    source_images = [
+                        SourceImageForSegmentation(
+                            s.ReferencedSOPClassUID,
+                            s.ReferencedSOPInstanceUID
+                        )
+                        for s in ref_series.ReferencedInstanceSequence
+                    ]
+                elif hasattr(ref_series, 'SeriesInstanceUID'):
+                    source_series = SourceSeriesForSegmentation(
+                        ref_series.SeriesInstanceUID
+                    )
+                    source_images = []
+                else:
+                    raise AttributeError(
+                        "The ReferencedSeriesSequence in the segmentation "
+                        "dataset does not contain the expected information"
+                    )
+            else:
+                raise AttributeError(
+                    "Dataset does not contain a ReferencedSeriesSequence "
```

Comment:
I would clarify that the **segmentation** dataset does not contain the attribute (similar to the above error message):

```suggestion
                    "Segmentation dataset does not contain a ReferencedSeriesSequence "
```

Reply:
Addressed in 4bea49d